domain-analysis/operations.py

```diff
@@ -0,0 +1,104 @@
+import arrow
+import os
+import whodap
+import validators
+import tldextract
+from tranco import Tranco
+from connectors.core.connector import get_logger, ConnectorError
+from .freq.freq import *
+
+
+logger = get_logger('domain_analysis')
+CACHE_DIR = '/tmp'
+FREQTABLE = os.path.dirname(os.path.realpath(__file__)) + '/freq/freqtable.freq'
+
+
+def analyze_domain(config, params):
+    try:
+        domain = params.get('domain')
+        fc = FreqCounter()
+        fc.load(FREQTABLE)
+        total_word_probability, average_probability = fc.probability(domain)
+        return {'Total Word Probability': total_word_probability, 'Average Probability': average_probability}
+
+    except Exception as exp:
+        logger.exception('Error Executing Code: {}'.format(exp))
+        raise ConnectorError('Error Executing Code: {}'.format(exp))
+
+def get_domain_popularity(config, params):
+    try:
+        domain = params.get('domain')
+        list_date = params.get('list_date', None)
+        list_date = arrow.get(list_date).format("YYYY-MM-DD") if list_date else None
+        tranco = Tranco(cache=True, cache_dir=CACHE_DIR)
```

Comment:
for how long does this file remain in /tmp? the top 1M list get refreshed daily. how does the connector action run on a subsequent day know that it has to refresh the file?

Reply:
new function _purge_old_files() to clean up old files, was under the impression the tranco package does it.